File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> v;
        v.push_back(0);
        v.push_back(1);
        if(n == 0) return {0};
        if(n == 1) return v;
        for(int i=2;i<=n;i++){
            if(i%2 == 0) {
                v.push_back(v[i/2]);
            } else{
                v.push_back(v[i/2] + 1);
            }
        }
        return v;
    }
};

File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] arr = new int[n+1];
        for(int i=0;i<=n;i++){
            int cnt = 0;
            int x = i;
            while(x != 0){
                x = x &(x-1);
                cnt++;
            }
            arr[i] = cnt;
        }
        return arr;
    }
}